gcp_airflow_foundations/source_class/ftp_sftp_source.py

```diff
@@ -0,0 +1,255 @@
+from dataclasses import fields
+from os import X_OK
+from urllib.parse import urlparse
+import logging
+from dacite import from_dict
+from dataclasses import dataclass
+from datetime import datetime
+import shutil
+import os
+import glob
+
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.dummy import DummyOperator
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.sensors.python import PythonSensor
+
+from gcp_airflow_foundations.operators.api.hooks.sftp_hook import SFTPHook
+from gcp_airflow_foundations.operators.api.sensors.sftp_sensor import SFTPFilesExistenceSensor
+from gcp_airflow_foundations.source_class.ftp_source import GenericFileIngestionDagBuilder
+from gcp_airflow_foundations.common.gcp.load_builder import load_builder
+from gcp_airflow_foundations.base_class.sftp_table_config import SFTPTableConfig
+
+
+class SFTPFileIngestionDagBuilder(GenericFileIngestionDagBuilder):
+    """
+    Builds DAGs to load files from an SFTP server to a BigQuery Table. Authenticates with the Airflow SFTP provider connection.
+
+    For specifying files to be uploaded, there are 3 options:
+        1) A metadata file is provided with templated daily file names (templating supports {{ ds }} and {{ TABLE_NAME }})
+        2) The field templated_file_name is provided in the table config (the same templating applies - but only 1 file can be provided in this manner)
+        3) A directory can be specified for .PARQUET files only in the table config
+
+    If a metadata file is provided, it can be a fixed file, or can be a new file supplied daily.
+    Airflow context variables are supported for the file naming, e.g.
+        TABLE_METADATA_FILE_{{ ds }}.csv
+    for a metadata file supplied daily.
+
+    The format of the metadata file should be a csv with one column as follows:
+        FILE_NAME_1
+        ...
+        FILE_NAME_N
+    with all files to ingest
+
+    If a directory for .PARQUET ingestion is provided, the field directory_prefix should point to the directory where
+    the outward-most external partition sits, and the field flag_file_path should point to the path of the _SUCCESS file on the sftp server.
+    A mod_time check occurs on the flag file to screen for daily ingestion on the sensor.
+    """
+    source_type = "SFTP"
+
+    def flag_file_sensor(self, table_config, taskgroup):
```

Comment:
I feel like most of this method could be in the base class. 
Would other file based success have dir_prefix and  flag_file_path? (I feel like yes)

Reply:
Yeah, this can move to the base class and the actual sensor implementation remain in the subclasses